configure.ac

```diff
@@ -241,24 +241,24 @@ AC_SEARCH_LIBS([clock_gettime], [rt],
 dnl check for libraries
 
 OMPI_CHECK_PORTALS4([portals4],
-    [transport="portals4"
-     transport_portals4="yes"],
+    [transport_portals4="yes"],
     [transport_portals4="no"])
 
 OPAL_CHECK_OFI([ofi],
-    [transport="ofi"
-     transport_ofi="yes"],
+    [transport_ofi="yes"],
     [transport_ofi="no"])
 
 # assure that only ofi is enabled if portals4 is also detected
-if test -n "$with_ofi" ; then
+if test "$transport_ofi" = "yes" ; then
```

Comment:
Yeah, we do need ```$with_X != "no"``` in these conditionals.  For example, if the user passes ```--without-ofi --with-portals4``` then ```$with_ofi``` is set to "no" and ```$with_portals4``` to "yes".  Without the check, this would choose OFI.